### PR TITLE
Expose AES-CMAC algorithm

### PR DESCRIPTION
**Pull request by xaqq**
Hello, 

This exposes `CKM_AES_CMAC` algorithm used in sign and verify operations. 
Added tests against SoftHSM2.

### CODE REVIEW

**Review by wiktor-k — APPROVED** [review]
I think it looks good 👍. Please fix the DCO thingy (I guess you can just `git commit --sign-off --amend --no-edit && git push --force`). Thanks!

File: cryptoki/tests/basic.rs [hunk 1]
```diff
@@ -1187,3 +1187,71 @@ fn ekdf_aes_cbc_encrypt_data() -> TestResult {
...
+        0x3c,
+    ];
+    let message: [u8; 16] = [
+        0x6b, 0xc1, 0xbe, 0xe2, 0x2e, 0x40, 0x9f, 0x96, 0xe9, 0x3d, 0x7e, 0x11, 0x73, 0x93, 0x17,
+        0x2a,
+    ];
+    let expected_mac: [u8; 16] = [
+        0x07, 0x0a, 0x16, 0xb4, 0x6b, 0x4d, 0x41, 0x44, 0xf7, 0x9b, 0xdd, 0x9d, 0xd0, 0x4a, 0x28,
+        0x7c,
+    ];
```
Comment (ionut-arm) [inline cryptoki/tests/basic.rs:1208]: I'm curious, are these test vectors from a spec?

**Review by ionut-arm — APPROVED** [review]
Looks good, thanks! 👍🏻

File: cryptoki/tests/basic.rs (hunk 1 above)
Comment (wiktor-k) [reply cryptoki/tests/basic.rs:1208]: If they are, it'd be good to note that in a comment. If they are not it'd be good to check the spec if there are any vectors in there :)

**Pull request merged**